### PR TITLE
Introduce run phase to compile and run javafx application

### PR DESCRIPTION
**Pull request by rfscholte**
With this PR it is possible to run the default/build lifecycle up until process-classes, followed by the execution of the application.
By specifying the `@Execute` on the mojo, you force Maven to first fork and execute that phase before executing itself. This only works when a goal is executed as part of a lifecycle (as it is forking the lifecycle). It won't have effect when `javafx:run` directly from the commandline, but by introducing the `run` phase (as part of the `run lifecycle`) this works.
Now you can execute `mvn run`, which will run the normal goals for a jar up until process-classes and optionally extra plugins if required.

This will require documentation, as this is quite unusual, but Maven supports this.

Most important reason: no compile magic in javafx:run, just configure as everybody expects.

### CODE REVIEW

File: src/main/java/org/openjfx/JavaFXRunMojo.java [hunk 1]
```diff
@@ -47,7 +50,7 @@
      * the PATH for the execution to work.
      * </p>
      */
-    @Parameter(property = "javafx.executable", defaultValue = "java")
+    @Parameter(property = "javafx.executable", defaultValue = "java.exe")
```
Comment (abhinayagarwal) [inline src/main/java/org/openjfx/JavaFXRunMojo.java]: This change seems to be specific to Windows and shouldn't be required.

**Comment by tiainen** [comment]
Thanks for contributing this PR.

As per the Contribution guidelines, please fill in the CLA so we can merge this PR once it has been approved.

**Comment by stephenc** [comment]
Note that this will not behave the way you might expect if you are in a multimodual checkout. You will need to just run on the specific module only as the other modules won’t understand the lifecycle

**Comment by jperedadnr** [comment]
The `mojo-executor` dependency can be removed now

**Comment by jperedadnr** [comment]
@rfscholte As a follow-up to this PR, I've filed #56. 

Mainly the goal is to include in the JavaFX plugin a "dependency" on the Maven compiler plugin 3.8.1, so users don't need to include this plugin in their pom, only to increase the default version.

As we discussed, this can be done by overriding the phases of the default lifecycle in the `components.xml` file, if I got it right.

Do you think that is the best way to accomplish this? 
Thanks!

**Pull request merged**